src/menu.cc

```diff
@@ -153,32 +147,45 @@ void _nm_menu_inject(void *nmc, QMenu *menu, nm_menu_location_t loc, int at) {
     if (before == nullptr)
         NM_LOG("it seems the original item to add new ones before was never actually added to the menu (number of items when the action was created is %d, current is %d), appending to end instead", at, actions.count());
 
+    NM_LOG("checking for config updates");
+    bool updated = nm_global_config_update(NULL); // if there was an error it will be returned as a menu item anyways (and updated will be true)
+    NM_LOG("updated = %d", updated);
+
     NM_LOG("checking for old items");
 
     for (auto action : actions) {
         if (action->property("nm_action") == true) {
-            return; // already added
-            /*menu->removeAction(action);
-            delete action;*/
+            if (!updated)
+                return; // already added items, up to date
```

Comment:
Is there a reason this isn't being checked before starting to iterate over the actions? Or is this your way if dealing with "first run"?

Reply:
Yes, that's my way of dealing with it.